pytorch_lightning/trainer/properties.py

```diff
@@ -507,7 +510,13 @@ def min_steps(self) -> Optional[int]:
 
     @property
     def callback_metrics(self) -> dict:
-        return self.logger_connector.callback_metrics
+        _cb_metrics: dict = self.logger_connector.callback_metrics
+        if not _cb_metrics and isinstance(self.accelerator.training_type_plugin, DDPSpawnPlugin):
```

Comment:
Then, I guess it should apply to the `TPUSpawn` Plugin as well.

Reply:
@kaushikb11 this won't be necessary, since TPUSpawn is derived from DDPSpawn and thereby is automatically included in this instance check :)

Reply:
This won't work as it will also trigger the warning when running inside the spawned processes which we want to avoid.
We need to check if torch `is_distributed`, as it won't be for the the main process.